blazor/datagrid/paging.md

```diff
@@ -7,443 +7,1162 @@ control: DataGrid
+    {
+        GridData = OrderData.GetAllRecords();
+    }
 
-                .disableFirst {
-                    pointer-events: none;
-                    opacity: 0.3;
-                }
+    private async Task CalculatePageSize(Syncfusion.Blazor.Inputs.ChangeEventArgs<int> args)
+    {
+        await Grid.GoToPageAsync(args.Value);
+        await Grid.Refresh();
+    }
+}
+{% endhighlight %}
+{% highlight c# tabtitle="OrderData.cs" %}
+public class OrderData
+    {
+        public static List<OrderData> Orders = new List<OrderData>();
 
-                .disableLast {
-                    pointer-events: none;
-                    opacity: 0.3;
-                }
+        public OrderData()
+        {
 
-                .disableFront {
-                    pointer-events: none;
-                    opacity: 0.3;
-                }
+        }
+        public OrderData(int? OrderID,string CustomerID,string ShipCity, string ShipName)
+        {
+            this.OrderID = OrderID;    
+            this.CustomerID = CustomerID;
+            this.ShipCity = ShipCity;
+            this.ShipName = ShipName;           
+        }
 
-                .disableBack {
-                    pointer-events: none;
-                    opacity: 0.3;
+        public static List<OrderData> GetAllRecords()
+        {
+            if (Orders.Count() == 0)
+            {
+               
+                int OrderID = 10248;
+
+                for (int i = 1; i < 3; i++)
+                {
+                    Orders.Add(new OrderData(OrderID+1, "VINET", "Reims", "Vins et alcools Chevali"));
+                    Orders.Add(new OrderData(OrderID+2, "TOMSP", "Münster", "Toms Spezialitäten"));
+                    Orders.Add(new OrderData(OrderID+3, "HANAR", "Rio de Janeiro", "Hanari Carnes"));
+                    Orders.Add(new OrderData(OrderID+4, "VICTE", "Lyon", "Victuailles en stock"));
+                    Orders.Add(new OrderData(OrderID+5, "SUPRD", "Charleroi", "Suprêmes délices"));
+                    Orders.Add(new OrderData(OrderID+6, "HANAR", "Lyon", "Hanari Carnes"));
+                    Orders.Add(new OrderData(OrderID+7, "CHOPS", "Rio de Janeiro", "Chop-suey Chinese"));
+                    Orders.Add(new OrderData(OrderID+8, "RICSU", "Münster", "Richter Supermarkt"));
+                    Orders.Add(new OrderData(OrderID+9, "WELLI", "Reims", "Wellington Import"));
+                   
+                    OrderID += 9;
                 }
-            </style>
-        </Template>
-    </GridPageSettings>
+            }
+            return Orders;
+        }
+
+        public int? OrderID { get; set; }
+        public string CustomerID { get; set; }
+        public string ShipCity { get; set; }
+        public string ShipName { get; set; }
+   }
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "https://blazorplayground.syncfusion.com/embed/VXLJXCiizsqxrUbb?appbar=false&editor=false&result=true&errorlist=false&theme=bootstrap5" %}
```

Comment:
it has 4 pages. but able to navigate till 3 alone. not able to go to 4th page

Reply:
Resolved